icu4c/source/i18n/calendar.cpp

```diff
@@ -1530,9 +1530,16 @@ void Calendar::computeGregorianFields(int32_t julianDay, UErrorCode& ec) {
         ec = U_ILLEGAL_ARGUMENT_ERROR;
         return;
     }
-    Grego::dayToFields(julianDay, fGregorianYear, fGregorianMonth,
-                       fGregorianDayOfMonth, gregorianDayOfWeekUnused,
-                       fGregorianDayOfYear, ec);
+    int32_t gregorianMonth, gregorianDayOfMonth, gregorianDayOfYear;
+    Grego::dayToFields(julianDay, fGregorianYear, gregorianMonth,
+                       gregorianDayOfMonth, gregorianDayOfWeekUnused,
+                       gregorianDayOfYear, ec);
+    U_ASSERT(gregorianMonth <= 12);
+    U_ASSERT(gregorianDayOfMonth <= 31);
+    U_ASSERT(gregorianDayOfYear <= 366);
+    fGregorianMonth = gregorianMonth;
+    fGregorianDayOfMonth = gregorianDayOfMonth;
+    fGregorianDayOfYear = gregorianDayOfYear;
```

Comment:
optional: Would it make sense to change Grego::dayToFields() to write to narrower values, so that you need not change the call site?